### PR TITLE
[Credo][Ycable] Fix for displaying 'N/A' firmware version when NIC endpoint is power off

### PR DESCRIPTION
**Pull request by xinyulin**
#### Description
This fix resolve the issue of being unable to obtain the firmware version of the self-side when the NIC end is power off.

#### Motivation and Context
To report the self-side firmware version for maintenance and trouble shooting. 

#### How Has This Been Tested?
Tested on Arista-7050CX3

The original version will show 'N/A' for all endpoints If the NIC endpoint is turned off, 
image

After applying the fix, the self-tor version can be displayed correctly.
image

### CODE REVIEW

**Comment by prgeor** [comment]
@xinyulin can you paste the CLI output with this fix

**Review by prgeor — APPROVED** [review]

**Comment by xinyulin** [comment]
> @xinyulin can you paste the CLI output with this fix

Sure!, added in the description.

File: sonic_y_cable/credo/y_cable_credo.py [hunk 1]
```diff
@@ -1463,14 +1463,21 @@ def get_firmware_version(self, target):
...
                     vsc_req_form = [None] * (YCable.VSC_CMD_ATTRIBUTE_LENGTH)
                     vsc_req_form[YCable.VSC_BYTE_OPCODE] = YCable.VSC_OPCODE_FWUPD
                     vsc_req_form[YCable.VSC_BYTE_OPTION] = YCable.FWUPD_OPTION_GET_INFO
                     status = self.send_vsc(vsc_req_form)
 
                     if status != YCable.MCU_EC_NO_ERROR:
-                        self.log_error('Get firmware version error (error code:0x%04X)' % (status))
-                        return None
+                        ''' should at least return local side fw version if nic is offline'''
+                        if status == 0x1A and ((read_side[0] == 4 and target == 1) or (read_side[0] == 2 and target == 2)):
```
Comment (vdahiya12) [inline sonic_y_cable/credo/y_cable_credo.py:1476]: can the read_side byr array be empty ?
if yes we could be causing exception here.

Comment (xinyulin) [reply sonic_y_cable/credo/y_cable_credo.py:1476]: Yes, I'd better to call get_read_side() to handle all the exceptions. I will submit a commit to fix it, thanks!

**Review by vdahiya12 — APPROVED** [review]

**Review by vdahiya12 — APPROVED** [review]

**Comment by vdahiya12** [comment]
LGTM, 
I have a generic feedback to have this suggestion for accessing byte_array/dict etc. to be error checked before accessing, on the entire module. 
Please let me know if you want more info on this.

File: sonic_y_cable/credo/y_cable_credo.py [hunk 2]
```diff
@@ -1469,8 +1475,12 @@ def get_firmware_version(self, target):
                     status = self.send_vsc(vsc_req_form)
 
                     if status != YCable.MCU_EC_NO_ERROR:
-                        self.log_error('Get firmware version error (error code:0x%04X)' % (status))
-                        return None
+                        ''' should at least return local side fw version if nic is offline'''
+                        if status == 0x1A and (read_side == target):
```
Comment (prgeor) [inline sonic_y_cable/credo/y_cable_credo.py:1479]: @xinyulin 0x1A can you name this status value

Comment (xinyulin) [reply sonic_y_cable/credo/y_cable_credo.py]: Sure, I made the correction in the last commit. Please review it.

**Review by prgeor — APPROVED** [review]

**Pull request merged**